tests/strings/static_string_check.cpp

```diff
@@ -1,13 +1,250 @@
+/**
+ * @file static_string_check.cpp
+ * @brief Unit Testing for StaticString class present in Wlib library
+ *
+ * @author Deep Dhillon
+ * @date November 4, 2017
+ * @bug No known bugs
+ */
 
 #include "gtest/gtest.h"
 #include "strings/StaticString.h"
 
-TEST(static_string_tests, constructor_tests){
+TEST(length_and_size, static_string_ctr_test){
     StaticString<8> string1{"helloooo"};    // text given
     StaticString<8> string2;                // no text
     StaticString<8> string3{string1};       // string object given
 
-    ASSERT_STREQ(string1.c_str(), "helloooo");
-    ASSERT_STREQ(string2.c_str(), "\0");
-    //ASSERT_STREQ(string3.c_str(), "helloooo");
-}
\ No newline at end of file
+    // check for length and size
+    EXPECT_TRUE(string1.capacity() == 8 && string1.length() == 8);
+    EXPECT_TRUE(string2.capacity() == 8 && string2.length() == 0);
+    EXPECT_TRUE(string3.capacity() == 8 && string3.length() == 8);

```

Comment:
Assert expression should be expressive.

```c++
ASSERT_EQ(8, string1.capacity());
ASSERT_EQ(8, string1.length());
```

instead of using only `EXPECT_TRUE`

Also remember that gtest uses 'Yoda' style which means it's `ASSERT_EQ(expected, actual)`

Reply:
An extra note, sometimes `ASSERT_EQ` funks out with overloaded operators, only then should you do `ASSERT_TRUE(val1 == val2)`